client/lib/sites-list/list.js

```diff
@@ -196,18 +193,11 @@ SitesList.prototype.update = function( sites ) {
 
 	this.markCollisions( sites );
 	this.data = sites.map( function( site ) {
-		var siteObj, result;
-
-		if ( sitesMap[ site.ID ] ) {
-			// Update existing Site object
-			siteObj = sitesMap[ site.ID ];
-			result = siteObj.set( site );
```

Comment:
>Could we maybe update `Site.prototype.set` to explicitly check for the icon case instead? I feel like this may cause regressions

The problem with this is that `set` _can_ be called with a partial set of attributes, e.g. plugin updates and plans after a purchase is completed.

Another option could be to normalize the received sites and assign `{ icon: undefined }` if unset so at least the key is passed through in the `set` object argument.